### PR TITLE
[FLINK-13529][table-planner-blink] Verify and correct agg function's semantic for Blink planner

### PR DESCRIPTION
**Pull request by beyond1920**
## What is the purpose of the change

Verify and correct agg function's semantic for Blink planner

## Brief change log

  - Refactor current ConcatAgg based on ANSI-SQL standard, including:
    * Rename concat_agg to ListAgg
    * replace default separator value from '\n' to ','
    * Exchange first parameter and second parameter position if there are two parameters.
    * add limitation that the second parameter must be character if there are two parameters.
    * Update effected UT and IT
  - Refactor first_value/last_value to remove second parameter.
  - Remove agg function APPROX_COUNT_DISTINCT, INCR_SUM in Blink planner

## Verifying this change
existed UT and IT
 
## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency):  no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/functions/sql/SqlFirstLastValueAggFunction.java [hunk 1]
```diff
@@ -45,7 +44,7 @@ public SqlFirstLastValueAggFunction(SqlKind sqlKind) {
 				sqlKind,
 				ReturnTypes.ARG0_NULLABLE_IF_EMPTY,
 				null,
-				OperandTypes.or(OperandTypes.ANY, OperandTypes.family(SqlTypeFamily.ANY, SqlTypeFamily.ANY)),
+				OperandTypes.ANY,
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/functions/sql/SqlFirstLastValueAggFunction.java]: Could you align the implementation with `org.apache.calcite.sql.fun.SqlFirstLastValueAggFunction`? The only different should be `requirsOver`, and would be great to mention it in the class javadoc.

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/AggFunctionFactory.scala [hunk 2]
```diff
@@ -610,19 +608,19 @@ class AggFunctionFactory(
...
     if (needRetraction(index)) {
-      new ConcatWithRetractAggFunction
+      new ListAggWithRetractAggFunction
     } else {
-      new ConcatAggFunction(1)
+      new ListAggFunction(1)
     }
   }
 
   private def createConcatWsAggFunction(
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/AggFunctionFactory.scala]: Rename the method.

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/AggFunctionFactory.scala [hunk 3]
```diff
@@ -610,19 +608,19 @@ class AggFunctionFactory(
       argTypes: Array[LogicalType],
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/AggFunctionFactory.scala]: Rename the method.

File: flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/functions/sql/FlinkSqlOperatorTable.java [hunk 4]
```diff
@@ -946,14 +946,9 @@ public SqlMonotonicity getMonotonicity(SqlOperatorBinding call) {
...
-	 * <code>CONCAT_AGG</code> aggregate function.
+	 * <code>LISTAGG</code> aggregate function.
 	 */
-	public static final SqlConcatAggFunction CONCAT_AGG = new SqlConcatAggFunction();
-
-	/**
-	 * <code>INCR_SUM</code> aggregate function.
-	 */
-	public static final SqlIncrSumAggFunction INCR_SUM = new SqlIncrSumAggFunction();
+	public static final SqlListAggFunction LISTAGG = new SqlListAggFunction();
```
[the remark this answers is not in the document]
Comment (beyond1920) [reply flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/functions/sql/FlinkSqlOperatorTable.java]: SqlStdOperatorTable.LISTAGG does not limit second parameter to constant character.
It's better to use our own `SqlListAggFunction`?

[the remark this answers is not in the document]
Comment (wuchong) [reply flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/functions/sql/FlinkSqlOperatorTable.java]: Yes. I agree. Please add a comment on `SqlListAggFunction` why we not use SqlStdOperatorTable.LISTAGG.

**Comment by beyond1920** [comment]
@wuchong Thanks for your review
I create [jira-13582](https://issues.apache.org/jira/browse/FLINK-13582) for improve  implementation of LISTAGG in release-1.10
And update codes, please have a look.

**Comment by wuchong** [comment]
Hi @beyond1920 , please do not squash commits & force-push before the reviewing is done. Otherwise, the reviewers have to go through the pull request entirely again...

**Comment by beyond1920** [comment]
@wuchong , I fix the compile error. Sorry to squash the commit in advance. I would squash the commit after the pr is ac and before merged later.

**Comment by wuchong** [comment]
Hi @beyond1920 , there are still some failed cases: https://travis-ci.org/wuchong/flink/jobs/568173514

**Comment by wuchong** [comment]
Travis passed. Merging...

**Comment by wuchong** [comment]
Travis passed. Merging...

**Pull request closed without merging**